test/build.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash -l
+#=======================================================================
+# Description:  This script runs a build test for the
+#               UFS Short-Range Weather App. The executables
+#               built are listed below in $executables_created.
+#               A pass/fail message is printed at the end of the output.
+#
+# Necessary input parameters: machine name (jet hera or cheyenne)
+#
+# Usage:  see function usage below
+#
+# Examples: ./build.sh $machine >& test.out &
+#
+set -eux    # Uncomment for debugging
+#=======================================================================
+
+fail() { echo -e "\n$1\n" >> ${TEST_OUTPUT} && exit 1; }
+
+function usage() {
+  echo
+  echo "Usage: $0 machine | -h"
+  echo
+  echo "       machine       [required] is one of: ${machines[@]}"
+  echo "       -h            display this help"
+  echo
+  exit 1
+}
+
+machines=( hera jet cheyenne )
+
+[[ $# -eq 0 ]] && usage
+if [ "$1" = "-h" ] ; then usage ; fi
+
+export machine=${1}
+machine=`echo "${machine}" | tr '[A-Z]' '[a-z]'`  # scripts in sorc need lower case machine name
+
+#-----------------------------------------------------------------------
+# Check that machine is valid
+#-----------------------------------------------------------------------
+if [[ "${machines[@]}" =~ "$machine" ]]; then
+  echo "machine ${machine} is valid"
+else
+  echo "ERROR: machine ${machine} is NOT valid"
+  exit 1
+fi
+
+#-----------------------------------------------------------------------
+# Set compilers to be tested depending on machine
+#-----------------------------------------------------------------------
+if [ "${machine}" == "cheyenne" ] ; then
+  compilers=( intel gnu )
+else
+  compilers=( intel )
+fi
+
+#-----------------------------------------------------------------------
+# Set some directories
+#-----------------------------------------------------------------------
+PID=$$
```

Comment:
Just for my enrichment, what does this syntax do?

Reply:
``$$`` is a process ID (PID) used to create a unique output filename each time the script is run so the previous output is saved.

Reply:
Thanks, @JulieSchramm!